### PR TITLE
Automate deployments (connect #1125)

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Before the PR (what is the issue or what needed to be done)
NA
#### The solution
+ Allows deployments using travis:
see instructions here https://github.com/akvo/akvo-flow-mobile/wiki/Deployments
+ Re-enables the android tests which were disabled by error
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [ ] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: app/src/androidTest/java/org/akvo/flow/activity/form/submittedformsview/FormActivityReadOnlyTest.java [hunk 1]
```diff
@@ -208,7 +210,7 @@ private void verifyQuestionView(Question question, int questionPosition) {
     private void verifyCaddisflyQuestionView(Question question) {
         String questionValue = getResponseValue(question);
 
-        List<CaddisflyTestResult> caddisflyTestResults = new CaddisflyJsonMapper()
+        List<CaddisflyTestResult> caddisflyTestResults = new CaddisflyJsonMapper(new Gson())
```
Comment (valllllll2000) [inline app/src/androidTest/java/org/akvo/flow/activity/form/submittedformsview/FormActivityReadOnlyTest.java]: Turns out tests were broken, had to fix it here or builds won't work

File: app/build.gradle [hunk 2]
```diff
@@ -85,17 +85,12 @@ android {
...
     buildTypes {
         buildTypes.each {
-            def serverBase = surveyProps['serverBase'].toString()
             def awsBucket = surveyProps['awsBucket'].toString()
+            def serverBase = "https://${awsBucket}.appspot.com"
             def awsAccessKeyId = surveyProps['awsAccessKeyId'].toString()
             def awsSecretKey = surveyProps['awsSecretKey'].toString()
             def apiKey = surveyProps['apiKey'].toString()
 
-            def signingKey = surveyProps['signingKey'].toString()
```
Comment (valllllll2000) [inline app/build.gradle]: Turns out signing key was not defined in any server config files, decided to remove it

File: .gitignore [hunk 3]
```diff
@@ -53,4 +53,6 @@ fastlane/report.xml
 fastlane/README.md
 build_config
 .env.default
-app/fabric.properties
\ No newline at end of file
+akvo-flow-server-config/
```
Comment (muloem) [inline .gitignore]: Does this mean we checkout the config folder within the `akvo-flow-mobile` folder?

File: .travis.yml [hunk 4]
```diff
@@ -29,21 +31,21 @@ before_install:
   - gem install fastlane --no-rdoc --no-ri --no-document --quiet
   - git remote set-url origin https://valllllll2000:$GITHUB_API_KEY@github.com/akvo/akvo-flow-mobile.git
```
Comment (muloem) [inline .travis.yml]: Maybe you can set the user as an environment variable as well?

File: app/build.gradle [hunk 5]
```diff
@@ -85,17 +85,12 @@ android {
 
     buildTypes {
         buildTypes.each {
-            def serverBase = surveyProps['serverBase'].toString()
             def awsBucket = surveyProps['awsBucket'].toString()
+            def serverBase = "https://${awsBucket}.appspot.com"
```
Comment (muloem) [inline app/build.gradle]: I would recommend to get this value from the `serverBase` as you had two lines above.  In the case of the older instances the `awsBucket` doesn't correspond to the `appId`

File: travis/deploy_release.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+if [ -z "$1" ]
+  then
+    echo "At least one instance needs to be provided"
+    exit 1
+fi
+
+[[ -n "${GITHUB_API_KEY}" ]] || { echo "GITHUB_API_KEY env var needs to be set"; exit 1; }
+
+git clone https://valllllll2000:$GITHUB_API_KEY@github.com/akvo/akvo-flow-server-config.git
```
Comment (muloem) [inline travis/deploy_release.sh]: Can you also set the user as a variable?

File: .gitignore (hunk 3 above)
Comment (valllllll2000) [reply .gitignore]: we clone it when deploying with travis

**Pull request merged**